minute/__init__.py

```diff
@@ -421,3 +422,50 @@ def libraries_unused_in_groups(libraries: List[Replicate], groups: List[ScalingG
                 else:
                     assert False, "Expected only Replicate or Pool"
     return list(unused)
+
+
+
+# Lander-Waterman equation that states:
+# C/X = 1 - exp( -N/X )
+# where
+# X = number of distinct molecules in library
+# N = number of read pairs
+# C = number of distinct fragments observed in read pairs
+def lander_waterman(x, c, n):
+    return c/x - 1 + math.exp(-n / x)
+
+# Picard procedure for estimating library size translated directly to python.
+# See Picard repository for the original: https://github.com/broadinstitute/picard
+#
+# Note that here we use as total/unique single ended fragments, because we deduplicate
+# based on 1st mate. However if we used proper-pairs the result would be very similar,
+# since in any case the paired-end read is a single sequencing event.
+def estimate_library_size(total_reads, duplicate_reads):
+    unique_reads = total_reads - duplicate_reads
+    if (total_reads > 0 and duplicate_reads > 0):
+        m = 1.0
+        M = 100.0
+
+        if (unique_reads >= total_reads or lander_waterman(m * unique_reads, unique_reads, total_reads) < 0):
+            print("Non valid values for pairs and unique pairs: {} {}".format(total_reads, unique_reads))
```

Comment:
This is not valid for the library size function but it can happen if very small BAM files show up and we end up having zeroes somewhere. So I thought it would be better not to fail the pipeline in that case. But maybe it is better practice to raise a `ValueError` and `try` on the `Snakefile` level.